### PR TITLE
Ability to create RTL variants of specific/all stories of a kind as substories

### LINKED ISSUE #1673: Chromatic QA checks with different storybook knob values
**Is your feature request related to a problem? Please describe.**
Currently, Chromatic QA checks just the standard set of props in Storybook. 

For examples such as the Consent Banner, the standard is using the `news` service. e.g. ?path=/story/consentbanner--default
It does not carry out tests against a service such as `arabic`, which has a different layout. It should be testing this variant as well. 

https://www.chromaticqa.com/component?appId=5d28eb5ee163f6002046d6fb&name=Components%7CConsentBanner&mode=interactive&buildNumber=289
Screen Shot consent banner chromatic QA

**Describe the solution you'd like**
There should be automated tests that check different variants of Storybook examples. 

**Describe alternatives you've considered**
Updating our stories, so we have a separate url for each variant. e.g. ConsentBanner RTL & ConsentBanner LTR.

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves https://github.com/bbc/psammead/issues/1673

**Overall change:** Changes `buildRTLSubstories` to create RTL variants of all stories of a specific kind or specific stories of specific kind. Previously `buildRTLSubstories` would create RTL variants of all stories which was considered not a good approach because this would double the length of time it takes for ChromaticQA to run in the CI pipeline which is something we are actively trying to reduce.

The original plan was to do this with a new decorator that we could decorate individual or many stories with however it's not possible to create new substories this way.

Examples of use:

**create RTL variants of all stories of a kind**
```jsx
import { buildRTLSubstories } from '@bbc/psammead-storybook-helpers';

buildRTLSubstories('Components|Paragraph');
```

**create RTL variants of specific stories of a kind**
```jsx
import { buildRTLSubstories } from '@bbc/psammead-storybook-helpers';

buildRTLSubstories('Components|Paragraph', {
  include: ['containing an inline link'],
});
```

This would produce new substories in Storybook:

Screenshot 2019-10-22 at 11 10 18

This new substory is then automatically added to ChromaticQA as the RTL example of the original story.

**Code changes:**

- updates `buildRTLSubstories.js` API and tests.
- updates changelog with new examples
- bumps major version number

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/utilities/psammead-storybook-helpers/src/buildRTLSubstories.js [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
-    stories.forEach(({ name, render }) => {
-      storiesOf(`${kind}/RTL`, module)
-        .addDecorator(withServicesKnob({ defaultService: 'arabic' }))
-        .add(`RTL - ${name}`, render);
+const matchesStoryKind = kind => story => story.kind === kind;
+
+const matchesStoryName = name => story => story.name === name;
+
+const buildRTLSubstory = (kind, name, storyFn) => {
+  const arabicServiceDecorator = withServicesKnob({ defaultService: 'arabic' });
```
Comment (thekp) [inline packages/utilities/psammead-storybook-helpers/src/buildRTLSubstories.js]: is it worth adding change this to:

```
  const rtlServiceDecorator = withServicesKnob({ defaultService: 'arabic', services:['arabic', 'persian', 'urdu', 'pashto'] });
```

**Review by ryanmccombe — APPROVED** [review]

File: packages/utilities/psammead-storybook-helpers/src/buildRTLSubstories.js (hunk 1 above)
Comment (jroebu14) [reply packages/utilities/psammead-storybook-helpers/src/buildRTLSubstories.js]: oh nice idea!

Comment (jroebu14) [reply packages/utilities/psammead-storybook-helpers/src/buildRTLSubstories.js]: is that all the rtl services?

Comment (thekp) [reply packages/utilities/psammead-storybook-helpers/src/buildRTLSubstories.js]: yup, I just confirmed on slack.

**Review by thekp — APPROVED** [review]
Great work, you just need to create follow up issues to apply these changes across our repos. 👍

**Comment by jroebu14** [comment]
@thekp I guess for Psammead we are still working on this issue https://github.com/bbc/psammead/issues/1673 although it doesn't detail which stories we should have rtl variants for. My guess is we should do them for components where the layout changes such as consent banner.

**Comment by thekp** [comment]
> do them for components where the layout changes such as consent banner.

This sounds right to me 👍

**Review by j-pendlebury — APPROVED** [review]
LGTM

**Pull request merged**